allennlp/common/util.py

```diff
@@ -106,12 +106,18 @@ def group_by_count(iterable: List[Any], count: int, default_value: Any) -> List[
 A = TypeVar("A")
 
 
-def lazy_groups_of(iterator: Iterator[A], group_size: int) -> Iterator[List[A]]:
+def lazy_groups_of(iterable: Iterable[A], group_size: int) -> Iterable[List[A]]:
     """
     Takes an iterator and batches the individual instances into lists of the
     specified size. The last list may be smaller if there are instances left over.
     """
-    return iter(lambda: list(islice(iterator, 0, group_size)), [])
+    iterable = iter(iterable)
```

Comment:
Nit: `iterator = iter(iterable)` The `islice` call below looks like a bug when operating on an arbitrary iterable.

Reply:
I like to keep the number of local variables down for debugging, but I see your point. Will make the change.